hyperspy/learn/ornmf.py

```diff
@@ -144,49 +156,84 @@ def __init__(self, rank, lambda1=1., kappa=1., store_r=False,
         store_r : bool
             If True, stores the sparse error matrix, False by default.
-        robust : bool
-            If True, the original OPGD implementation is used for
-            corruption/outlier regularization, otherwise L2-norm. False by
-            default.
+        method : {None, 'PGD', 'RobustPGD', 'MomentumSGD'}
+            'PGD' - Proximal gradient descent
+            'RobustPGD' - Robust proximal gradient descent
+            'MomentumSGD' - Stochastic gradient descent with momentum
+            If None, set to PGD
+        subspace_learning_rate : float
+            Learning rate for stochastic gradient descent.
+        subspace_momentum : float
+            Momentum parameter for stochastic gradient descent. Must be in
+            range 0 <= x <= 1.
 
         """
-        self.robust = robust
+        self.robust = False
+        self.subspace_tracking = False
         self.nfeatures = None
         self.rank = rank
         self.lambda1 = lambda1
         self.kappa = kappa
         self.H = []
+        self.t = 0
+
+        if method is None:
+            _logger.warning(
+                "No method specified. Defaulting to "
+                "'PGD' (proximal gradient descent)"
+            )
+            method = "PGD"
+
+        if method not in ("PGD", "RobustPGD", "MomentumSGD"):
+            raise ValueError("'method' not recognised")
+
+        if method == "RobustPGD":
+            self.robust = True
+
+        if method == "MomentumSGD":
+            self.subspace_tracking = True
+            if subspace_momentum < 0.0 or subspace_momentum > 1:
+                raise ValueError('"subspace_momentum" must be in range [0, 1]')
+
+        self.subspace_learning_rate = subspace_learning_rate
+        self.subspace_momentum = subspace_momentum
+
         if store_r:
             self.R = []
         else:
             self.R = None
 
     def _setup(self, X, normalize=False):
-        self.h, self.r = None, None
+        self.h, self.r, self.v = None, None, None
         if isinstance(X, np.ndarray):
             n, m = X.shape
             if normalize:
                 self.X_min = X.min()
                 self.X_max = X.max()
                 self.normalize = normalize
-                # actually scale the data to be between 0 and 1, not just close
-                # to it..
+                # actually scale the data to be between 0 and 1,
+                # not just close to it..
                 X = _normalize(X, ar_min=self.X_min, ar_max=self.X_max)
-                # X = (X - self.X_min) / (self.X_max - self.X_min)
+
             avg = np.sqrt(X.mean() / m)
         else:
             if normalize:
-                _logger.warning("Normalization with an iterator is not"
-                                " possible, option ignored.")
+                _logger.warning(
+                    "Normalization with an iterator is not" " possible, option ignored."
```

Comment:
Too many "" --> Should be: "Normalization with an iterator is not possible, option ignored."